mesonbuild/scripts/dist.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2017 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os, sys
+import shutil
+import argparse
+import subprocess
+import pickle
+import hashlib
+import tarfile, zipfile
+import tempfile
+from glob import glob
+from mesonbuild.environment import detect_ninja
+
+def create_hash(fname):
+    hashname = fname + '.sha256sum'
+    m = hashlib.sha256()
+    m.update(open(fname, 'rb').read())
+    with open(hashname, 'w') as f:
+        f.write('%s %s\n' % (m.hexdigest(), os.path.split(fname)[-1]))
+
+def create_zip(zipfilename, packaging_dir):
+    prefix = os.path.split(packaging_dir)[0]
+    removelen = len(prefix) + 1
+    with zipfile.ZipFile(zipfilename,
+                         'w',
+                         compression=zipfile.ZIP_DEFLATED,
+                         allowZip64=True) as zf:
+        zf.write(packaging_dir, packaging_dir[removelen:])
+        for root, dirs, files in os.walk(packaging_dir):
+            for d in dirs:
+                dname = os.path.join(root, d)
+                zf.write(dname, dname[removelen:])
+            for f in files:
+                fname = os.path.join(root, f)
+                zf.write(fname, fname[removelen:])
+
+def del_gitfiles(dirname):
+    for f in glob(os.path.join(dirname, '.git*')):
```

Comment:
So, I was thinking, `.gitignore` or `.gitattributes` are allowed to be in any subdirectory. I don't think this deletes them; should it be doing so?